GVFS/GVFS.Virtualization/Projection/FileTypeAndMode.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+namespace GVFS.Virtualization.Projection
+{
+    public struct FileTypeAndMode
```

Comment:
Do we need this whole new class instead of just using the ushort and having some method to check the ushort for values?

Reply:
> Do we need this whole new class instead of just using the ushort and having some method to check the ushort for values?

There was a suggestion in a previous round of PR feedback that we use a more explicit type for type+mode.  

@sanoursa, after seeing how the code looks with a wrapper `struct` do you have a preference on sticking with the struct or just providing helper methods for the `ushort`?

Reply:
I think it depends on the cost of masking out the bits of the ushort. Right now we're doing it twice - once in the index parsing code, and then again in the provider. If that cost doesn't matter, than storing a ushort with helper methods could suffice, though I think a new class makes the code cleaner.

Reply:
> If that cost doesn't matter, than storing a ushort with helper methods could suffice, though I think a new class makes the code cleaner.

The cost of masking is very low, and it's also true that we won't add much memory overhead to store the values separately.

I can switch to applying the mask once and having our own values for type (rather than re-using the values from git).  If in the future we need the memory back we can switch to using a `ushort`.